### ISSUE TITLE
Question about the training of MatchingNetwork

### ISSUE THREAD

**Issue by tonysy** [open-mmlab/mmfewshot#20]
Hi, Great Job.

I have some questions about the training process of the matching network(classification)

- In this line, https://github.com/open-mmlab/mmfewshot/blob/31583cccb8ef870c9e688b1dc259263b73e58884/configs/classification/matching_net/mini_imagenet/matching-net_conv4_1xb105_mini-imagenet_5way-1shot.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L28
You use num_shots=5 for training 5-way-1-shot, is this a bug?
- The batch size shown in the result table is 64, I would like to know whether this number is the training batch size or test batch size?
- How many gaps between the meta-val and meta-test split in your experiment?
   - In the log of matching_net 5-way-1-shot, the max accuracy is about 51%, while the test result is 53%, does it means there exists ~2 points between two sets? 

Thanks, 
Best

**Comment by linyq17** [comment]
1. The `num_shots` for training and meta testing settings are independent. During training, any number of shots can be used. Currently, we use the same training setting for all metric-based models under 1 shot and 5 shot settings.
2. The batch size is training batch size. Thanks for pointing the wrong batch size, we will correct the readme ASAP.
3. The gap is highly related to datasets. Also, it can be different and unstable for different models and settings since we only use 100 validation episodes.

**Comment by tonysy** [comment]
- For the first question, I would like to share some ideas.

From my understanding, it's related to the definition of the `task` or `episode`, different shots may influence the equivalent tasks(or episodes), typically one task consists of `num_support_way(5)*num_support_shots(1)+num_query_shots(1)=6` .

There always exist implementation differences in different public codebases, and it's hard to conduct a fair comparison between them. Hope this project could be a common benchmark for future research. It would be great if you can provide a more detailed illustration of the configuration(e.g., the definition of one task, how many equivalent tasks in total, definition of the batch size under the meta-learning setting, cross-comparison between different methods in one table) in the readme.

**Comment by linyq17** [comment]
Thanks for your suggestions.
1. Indeed, it is hard to define the most common practice among different public codebases. From my experience, a larger batch size (more ways or more shots) empirically leads to better results in most settings.
2. Sure, we will consider adding more illustrations in the readme and docs. The illustration of configuration can be found in [here ](https://mmfewshot.readthedocs.io/en/latest/classification/customize_config.html#an-example-of-baseline).
